app/react/Layout/TableViewer.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+
+import { connect } from 'react-redux';
+import { TableRow } from 'app/Library/components/TableRow';
+
+export interface DocumentViewerProps {
+  rowListZoomLevel: number;
+  documents: any;
+  storeKey: 'library' | 'uploads';
+  clickOnDocument: (...args: any[]) => any;
+  onSnippetClick: (...args: any[]) => any;
+  deleteConnection: (...args: any[]) => any;
+  search: any;
+  templates: any;
+  thesauris: any;
+  columns: any[];
+}
+
+class TableViewerComponent extends Component<DocumentViewerProps> {
+  render() {
+    const columns = this.props.columns.filter(c => !c.get('hidden'));
+    return (
+      <div className="tableview-wrapper">
+        <table>
+          <thead>
+            <tr>
+              {columns.map((column: any, index: number) => (
+                <th className={!index ? 'sticky-col' : ''} key={index}>
+                  <div className="table-view-cell">{column.get('label')}</div>
+                </th>
+              ))}
+            </tr>
+          </thead>
+          <tbody>
+            {this.props.documents.get('rows').map((document: any, index: number) => (
+              <TableRow
+                {...{
+                  document,
+                  columns,
+                  key: index,
+                  onClick: this.props.clickOnDocument,
+                  storeKey: this.props.storeKey,
+                  templates: this.props.templates,
+                  thesauris: this.props.thesauris,
```

Comment:
As mentioned above, there is no need to pass this over. They can be fetched in TableRow in map state to props.